client/app/src/main/java/com/microsoft/research/karya/injection/DaoModule.kt

```diff
@@ -45,4 +48,22 @@ class DaoModule {
   fun provideMicroTaskAssignmentDao(karyaDatabase: KaryaDatabase): MicroTaskAssignmentDao {
     return karyaDatabase.microtaskAssignmentDao()
   }
+
+  @Provides
+  @Reusable
+  fun provideKaryaFileDao(karyaDatabase: KaryaDatabase): KaryaFileDao {
+    return karyaDatabase.karyaFileDao()
+  }
+
+  @Provides
+  @Reusable
+  fun provideMicroTaskAssignmentDaoExtra(karyaDatabase: KaryaDatabase): MicrotaskAssignmentDaoExtra {
+    return karyaDatabase.microtaskAssignmentDaoExtra()
+  }
+
+  @Provides
+  @Reusable
+  fun provideMicroTaskDaoExtra(karyaDatabase: KaryaDatabase): MicrotaskDaoExtra {
+    return karyaDatabase.microtaskDaoExtra()
+  }
```

Comment:
I don't like the idea of having separate `Dao` and `DaoExtra`. Just move all the methods into their respective parent daos.

Reply:
DaoExtra contains the methods that do not do basic operations and are somewhat use-case specific. Separating them would clearly show what methods are for certain use-cases and what methods for general interaction

Reply:
The reason for using Dao is to keep all the things specific to one table in a single object. I don't see any reason to import both `Dao` and `DaoExtra` just to perform one operation. Also most of the `Dao` objects do not have a lot of functions so adding a few more won't hurt.